### PR TITLE
Feature/e2e test ci

### PR DESCRIPTION
**Pull request by fmoura**
Add End to End Tests to CI

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -126,8 +126,58 @@ jobs:
       - name: Run tests
         run: cargo test
 
+  build-ci-base :
```
Comment (gligneul) [inline .github/workflows/build.yml:129]:
```suggestion
  build-ci-base:
```

File: build/docker-bake.hcl [hunk 2]
```diff
@@ -45,3 +45,10 @@ target "rollups-node-devnet" {
   inherits = ["common"]
   target   = "rollups-node-devnet"
 }
+
+target "rollups-node-ci-base" {
+  inherits = ["common"]
+  target   = "rollups-node-ci-base"
+  dockerfile = "./Dockerfile"
+  context  = "."
```
Comment (gligneul) [inline build/docker-bake.hcl:53]: Why overwrite these?

File: build/Dockerfile [hunk 3]
```diff
@@ -331,3 +331,23 @@ USER cartesi
 
 # Set the Go supervisor as the command.
 CMD [ "cartesi-rollups-node" ]
+
+
+# STAGE: rollups-ci-base
+#
+# This stage prepares the base CI image
+FROM debian:bookworm as rollups-node-ci-base
```
Comment (gligneul) [inline build/Dockerfile:339]:
```suggestion
FROM ${BASE_IMAGE} as rollups-node-ci-base
```

I would use the base image to be consistent.

File: build/Dockerfile [hunk 4]
```diff
@@ -331,3 +331,23 @@ USER cartesi
...
+
+# Install git
+RUN apt update &&  apt install -y git-all
+
+# Install docker
+RUN <<EOF
+    apt update && apt install -y --no-install-recommends ca-certificates curl
```
Comment (gligneul) [inline build/Dockerfile:346]: Why these two commands are separate?

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -152,6 +202,9 @@ jobs:
         uses: actions/setup-go@v5
         with:
           go-version-file: 'go.mod'
+      
+      - name: Fix VCS Linter Issue
```
Comment (gligneul) [inline .github/workflows/build.yml:206]: What is VCS? What is the issue?

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -126,8 +126,58 @@ jobs:
...
+      - name: Build and push docker image
+        id: docker_build
+        uses: depot/bake-action@v1
+        with:
+          files: |
+            ./docker-bake.hcl
+            ${{ steps.docker_meta.outputs.bake-file }}
+            ./docker-bake.platforms.hcl
+          targets: rollups-node-ci-base
+          push: true
```
Comment (gligneul) [inline .github/workflows/build.yml:166]: You should add this image to the clean-up job.
https://github.com/cartesi/rollups-node/blob/74d4381f84fb68ee9d6aa2baa9a3bff436b34b3f/.github/workflows/clean-up-images.yml#L19

**Review by gligneul — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 5 above)
Comment (fmoura) [reply .github/workflows/build.yml:206]: Go Linter tries to check some "build stamping" against some information from the VCS, but fails reading this information.

File: build/Dockerfile (hunk 4 above)
Comment (fmoura) [reply build/Dockerfile]: Not really needed.

**Review by gligneul — APPROVED** [review]

**Pull request merged**